src/scripts/scripts/zone/hellfire_citadel/blood_furnace/boss_broggok.cpp

```diff
@@ -172,13 +179,35 @@ struct boss_broggokAI : public ScriptedAI
         }
     }
 
+    //This is called either per instance ID, or per player for that instance ID.
+    //It calls this even if you're out of instance - as long as you have an ID.
+    //When you reset instance, it stops calling it.
+    //Then: The very moment you enter the dungeon, it begins calling it.
+    //It's called about 5 times per second.
+    //It STOPS executing this method once the boss dies.
     void UpdateAI(const uint32 diff)
     {
+        //Save processing power by preventing many if statements being called.
+        if (phase == EVENT_NULL)
+            return;
+        else
+        {
+            //Here is a fix to a bug. There was no way to reset the boss gauntlet packs if you wiped on it.
+            //Therefore, when you run back, encounter is still in progress, and you can simply pull next pack.
+            //They should reset.
+            if (pMap->GetAlivePlayersCountExceptGMs() == 0)
```

Comment:
will be ways more better than now, but it still won't work with feign death/vanish/pala bubble